components/future_ext/src/retry.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022-2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! Util function to retry future.
+
+use std::time::Duration;
+
+use futures::Future;
+
+pub struct RetryConfig {
+    pub max_retries: usize,
+    pub interval: Duration,
+}
+
+impl Default for RetryConfig {
+    fn default() -> Self {
+        Self {
+            max_retries: 3,
+            interval: Duration::from_millis(500),
+        }
+    }
+}
+
+pub async fn retry_async<F, Fut, T, E>(f: F, config: &RetryConfig) -> Fut::Output
+where
+    F: Fn() -> Fut,
+    Fut: Future<Output = Result<T, E>>,
+{
+    for _ in 0..config.max_retries - 1 {
```

Comment:
What will happen if `max_retries` less than 0? do we need to validate it?

Reply:
when max_retries is 0, the for loop won't get executed, so no retry happens.  